fix: warn in case of unexpected terramate block

i4ki opened the pull request:
```
$ cat stack-a/terramate.tm.hcl
stack {
	name = "awesome-stack-a"
}

terramate {

}
$ terramate list
2022-08-04T02:17:37+01:00 WRN  error="/home/i4k/src/mineiros/tm-example/stack-a/terramate.tm.hcl:5,1-10: `terramate` block is only allowed at the project root directory" action=checkConflicts()
stack-a
stack-b
stack-c
```

katcipis commented on hcl/hcl.go:
```diff
@@ -190,6 +191,7 @@ type TerramateParser struct {
 	// parsedFiles stores a map of all parsed files
 	parsedFiles map[string]parsedFile
 
+	strict bool
```
JS feelings :laughing:

katcipis commented on hcl/hcl.go:
```diff
@@ -1406,6 +1423,48 @@ func (p *TerramateParser) parseTerramateSchema() (Config, error) {
+			errors.E(ErrConfigConflict, stackblock.TypeRange,
+				"`stack` block conflicts with `terramate` block defined at %s",
+				tmblock.RawOrigins[0].TypeRange.String(),
+			),
+
+			errors.E(ErrConfigConflict, tmblock.RawOrigins[0].TypeRange,
```
```suggestion
			),
			errors.E(ErrConfigConflict, tmblock.RawOrigins[0].TypeRange,
```

katcipis commented on hcl/hcl_stack_test.go:
```diff
@@ -330,7 +352,8 @@ func TestHCLParserStack(t *testing.T) {
 		},
 		{
-			name: "after: empty set works",
+			name:      "after: empty set works",
```
```suggestion
			name: "after: empty set works",
```

katcipis commented on hcl/hcl_stack_test.go:
```diff
@@ -160,7 +181,8 @@ func TestHCLParserStack(t *testing.T) {
 		},
 		{
-			name: "empty name",
+			name:      "empty name",
```
```suggestion
			name: "empty name",
```

I think the code is not formatted no ? :thinking:

i4ki commented:
@katcipis I changed to report an error if `terramate` is not at the rootdir because my previous assumption was not valid, the docs still allow `stack + terramate` in the same dir **if it's root dir**.
The change broke some tests because `cfg.Terramate` was the easiest one to test the merging at the config level. I removed those testcases because we still have lots of tests for globals + import covering the same merge thing, so I think this is fine.

katcipis commented on hcl/hcl.go:
```diff
@@ -1406,6 +1423,39 @@ func (p *TerramateParser) parseTerramateSchema() (Config, error) {
 
+func (p *TerramateParser) checkConfigSanity(cfg Config) error {
+	logger := log.With().
+		Str("action", "TerramateParser.checkConfigSanity()").
```
Noice
